File: jmix-templates/content/project/application/src/test/java/${project_rootPath}/test_support/UiIntegrationTest.java
package ${project_rootPackage}.test_support;

import ${project_rootPackage}.${project_classPrefix}Application;
import com.vaadin.flow.component.Component;
import io.jmix.flowui.ViewNavigators;
import io.jmix.flowui.component.UiComponentUtils;
import io.jmix.flowui.testassist.FlowuiTestAssistConfiguration;
import io.jmix.flowui.testassist.UiTest;
import io.jmix.flowui.testassist.UiTestUtils;
import io.jmix.flowui.view.View;
import org.junit.jupiter.api.Assertions;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.Optional;

@UiTest
@SpringBootTest(classes = {${project_classPrefix}Application.class, FlowuiTestAssistConfiguration.class})
public abstract class UiIntegrationTest {

    @Autowired
    ViewNavigators viewNavigators;

    protected ViewNavigators getViewNavigators() {
        return viewNavigators;
    }

    /**
     * Returns instance of currently navigated view.
     */
    protected <T extends View<?>> T getCurrentView() {
        return UiTestUtils.getCurrentView();
    }

    /**
     * Returns a component defined in the screen by the component id.
     * Throws an exception if not found.
     */
    @SuppressWarnings("unchecked")
    protected <T> T findComponent(View<?> view, String componentId) {
        Optional<Component> component = UiComponentUtils.findComponent(view, componentId);
        Assertions.assertTrue(component.isPresent());
        return (T) component.get();
    }
}

File: jmix-templates/content/project/application-kotlin/src/test/kotlin/${project_rootPath}/test_support/UiIntegrationTest.kt
package ${project_rootPackage}.test_support

import ${project_rootPackage}.${project_classPrefix}Application
import io.jmix.flowui.ViewNavigators
import io.jmix.flowui.component.UiComponentUtils
import io.jmix.flowui.testassist.FlowuiTestAssistConfiguration
import io.jmix.flowui.testassist.UiTest
import io.jmix.flowui.testassist.UiTestUtils
import io.jmix.flowui.view.View
import org.junit.jupiter.api.Assertions
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@UiTest
@SpringBootTest(classes = [${project_classPrefix}Application::class, FlowuiTestAssistConfiguration::class])
abstract class UiIntegrationTest {

    @Autowired
    lateinit var viewNavigators: ViewNavigators

    /**
     * Returns instance of currently navigated view.
     */
    protected fun <T : View<*>> getCurrentView(): T {
        return UiTestUtils.getCurrentView()
    }

    /**
     * Returns a component defined in the screen by the component id.
     * Throws an exception if not found.
     */
    @Suppress("UNCHECKED_CAST")
    protected fun <T> findComponent(view: View<*>, componentId: String): T {
        val component = UiComponentUtils.findComponent(view, componentId)
        Assertions.assertTrue(component.isPresent)
        return component.get() as T
    }
}